source/slang/slang-ir-layout.cpp

```diff
@@ -234,7 +234,7 @@ case kIROp_##TYPE##Type:                                        \
         auto anyValType = cast<IRAnyValueType>(type);
         outSizeAndAlignment->size = getIntVal(anyValType->getSize());
         outSizeAndAlignment->alignment = 4;
-        *outSizeAndAlignment = rules->alignCompositeElement(*outSizeAndAlignment);
+        *outSizeAndAlignment = rules->alignCompositeElementOfNonAggregate(*outSizeAndAlignment);
```

Comment:
This eventually is lowered into a struct so it need to be consistent with struct type case.